src/runCompletion.ts

```diff
@@ -48,3 +49,23 @@ function getMaxResults(): number {
 
   return MAX_NUM_RESULTS;
 }
+
+export type KnownLanguageType = keyof typeof languages;
+
+export function getLanguageFileExtension(
+  languageId: string
+): string | undefined {
+  return languages[languageId as KnownLanguageType];
+}
+
+export function getFileNameWithExtension(document: TextDocument): string {
+  const { languageId, fileName } = document;
+  let filenameWithExtension = fileName;
```

Comment:
why is that needed?

Reply:
Current binary API infers the file's language using its full name including the file extension. Untitled files have no file extension to their name. That way we can add the file extension manually to the untitled file's name.